[14.0] shopfloor, delivery: force qty to process to 1 when scanning a product or a lot

sebalix opened the pull request:
Ref. rau-51

sebalix committed:
shopfloor, delivery: force qty to process to 1 when scanning a product

TDu committed:
shopfloor, delivery: change processed quantiy on product scan

Until now a line was always fully delivered.

With this change

For move lines without pacakge:

* When scanning a product only a quantity of one should be processed.
* When scanning a packaging process the quantity of the packaging.

For move lines with a package:

* When scanning a product only if the line package contains one unity of
  the product and that product only.

jbaudoux commented on shopfloor/actions/message.py:
```diff
@@ -370,6 +370,15 @@ def product_mixed_package_scan_package(self):
+        return {
+            "message_type": "warning",
+            "body": _(
+                "This product is present multiple times in a package, "
```
```suggestion
                "This product is part of a package, "
```

sebalix replied:
fixed

sebalix replied:
fixed

sebalix commented on shopfloor/models/stock_quant_package.py:
```diff
@@ -92,3 +92,7 @@ def _move_package_quant_move_values(self, quant, location, quantity):
             "owner_id": quant.owner_id.id,
         }
+
+    def has_one_product(self):
+        self.ensure_one()
+        return len(self.quant_ids.product_id) == 1
```
Note to myself: this method is not used anymore, to remove

sebalix committed:
shopfloor, delivery: rework 'Process as pre-packaged' option

As the tooltip of the option was describing, its main goal is to
validate the delivery order (and create a backorder if needed) as soon
as a goods is scanned. This could be a package, a product, a packaging
or a lot.

Also, in all cases, when we scan a lot we want to process 1 qty (like
for product scanning), not all the line.

TDu commented:
LG, not sure about that pylint error...

simahawk commented:
for the pre-commit issue: please open another pr after running `copier -f update`.

simahawk reviewed (approved):
LG

sebalix commented:
/ocabot merge minor

sebalix commented:
/ocabot merge minor
